### PR TITLE
fix #181811 validate instrument text name after editing

### PR DESCRIPTION
**Pull request by ericfont**
I've tested this and verified that the validateText(s) I inserted here properly changes an '&' into an 'amp;'

### CODE REVIEW

File: libmscore/iname.cpp [hunk 1]
```diff
@@ -73,6 +73,11 @@ void InstrumentName::endEdit()
 
       QString s = plainText();
 
+      if (!validateText(s)) {
+            qWarning("The instrument name is invalid.");
```
Comment (Jojo-Schmitz) [inline libmscore/iname.cpp]: Maybe `qWarning("Invalid instrument name: <%s>", s.toUtf8().data());`?

**Comment by ericfont** [comment]
I've sucessfully saved as mscz after inserting the '&' and I've verified that I can open the file again and I see the & displayed as instrument text.

File: libmscore/iname.cpp (hunk 1 above)
Comment (ericfont) [inline libmscore/iname.cpp]: sure...let me repush.

**Comment by Jojo-Schmitz** [comment]
should go to master and 2.1

**Comment by ericfont** [comment]
@Jojo-Schmitz I just saw your message on IRC: does it fix <smb>flat</sym> ?

But I didn't test that yet...I have already switched my code base back to master...I will have to swap back in  an hour or so to test what you suggested...

**Comment by Jojo-Schmitz** [comment]
It should fix that too, along with bold, italic etc.

**Comment by ericfont** [comment]
Yeah, I would expect validateText would do all sorts of such fixups.

**Comment by lasconic** [comment]
Clicking on an instrument name causes a crash for me in master. So I don't manage to test this PR. I guess it can't harm...

**Comment by ericfont** [comment]
@lasconic yes.  as you might recall there were a few different crashes I discovered with clicking on instrument names, which is a seperate issue I have no idea how to fix.  But this PR is still needed and will do no harm.

**Pull request merged**